### DISCUSSION TITLE
Skip output 'AWS_ACCOUNT_ID' since it may contain secret

### DISCUSSION THREAD

**Discussion by jonathan-be21** [community/community#26636] [Actions]
<p>Hi everyone</p>
<p>we’re using multiple “::set-output” statements in order to pass vars between jobs.</p>
<p>previously everything worked perfectly, but since we’ve changed some Var’s values - we’re receiving the following warning:<br>
<img src="https://user-images.githubusercontent.com/65497025/181098977-f0b2074a-8054-4682-a8ad-bf34cb012696.png" alt="image" data-base62-sha1="nQSM6r8QTYfbuwv6fib7WPnVSs" width="589" height="171"></p>
<p>the VAR isn’t holds any secret value so i have no problem with showing it up, but we can’t use github secret instead as we the value is dynamic.</p>
<p>how should we tell github to stop treating those values as a secret?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jonathan-be21">@jonathan-be21</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>This is by designed.  Once you change the secrets value and it becomes part of the var value, the warning will happen, it’s automatically by github action and cannot be avoided.</p>
<p>As the <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjobs_idoutputs" rel="nofollow noopener">doc</a> mentioned:</p>
<p>Job outputs are strings, and job outputs containing expressions are evaluated on the runner at the end of each job.  <strong>Outputs containing secrets are redacted on the runner and not sent to GitHub Actions</strong> .</p>
<p>Thanks</p>

**Comment by jonathan-be21** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a>,</p>
<p>i wasn’t talking about using a github secret. i’m using a simple env var that github actions <em>thinks</em> it’s a secret.</p>
<p>it’s not a secret value by any mean (it’s just a long number actually)</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jonathan-be21">@jonathan-be21</a>,</p>
<p>Thanks for your reply!</p>
<p>Sorry i’m a little confused, could you please provide a sample code for further investigation?</p>
<p>If your var value is a long number, for example: 123456789, and using a secret (value set as 456) other place in the workflow, the warning will happen. The job output cannot get the value.</p>
<p>I repro the error on my side:<br>
My workflow file: <a href="https://github.com/weide-zhou/ticket13/actions/runs/176889434/workflow" rel="nofollow noopener">https://github.com/weide-zhou/ticket13/actions/runs/176889434/workflow</a><br>
Workflow run: <a href="https://github.com/weide-zhou/ticket13/runs/893614256?check_suite_focus=true#step:3:19" rel="nofollow noopener">https://github.com/weide-zhou/ticket13/runs/893614256?check_suite_focus=true#step:3:19</a></p>
<p>Thanks</p>

**Comment by jonathan-be21** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> thank you for your help so far.</p>
<pre><code class="lang-auto">jobs:
  build:
    runs-on: ubuntu-latest
    outputs:
      AWS_ACCOUNT_ID:      ${{ steps.retrieve_secrets.outputs.AWS_ACCOUNT_ID }}
      ECR_REPO_URI:        ${{ steps.retrieve_secrets.outputs.ECR }}        
    
    steps:      
      - name: "Checkout current repo"
        uses: actions/checkout@v2

      - name: "Configure AWS credentials"
        uses: aws-actions/configure-aws-credentials@v1

      - name: "Retrieve Secrets"
        id: retrieve_secrets
        shell: bash        
        run: |
          # Get secrets
          aws secretsmanager get-secret-value --secret-id $SECRET &gt; ENV_VARS.json
          echo "::set-output name=AWS_ACCOUNT_ID::$(cat ENV_VARS.json | jq -r '.AWS_ACCOUNT_ID')"
          echo "::set-output name=ECR::$(cat ENV_VARS.json | jq -r '.ECR')"
</code></pre>
<p>As you can see, I’ve never declared a secret but still receive the above error</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jonathan-be21">@jonathan-be21</a>,</p>
<p>What’s display in the <code>log</code> for step  “Retrieve Secrets”? Does it contain star ‘*’?  like below:<br>
<code>echo "::set-output name=AWS_ACCOUNT_ID::1212**3434 </code></p>
<p>Is there any secrets in your workflow other places? And how you define <code>$SECRET</code> in the yaml?</p>
<p>Thanks</p>

**Comment by jonathan-be21** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> no, the display shows only:<br>
##[warning]Skip output ‘AWS_ACCOUNT_ID’ since it may contain secret</p>
<p>I’m working around it by setting an hardcoded github secret, but even if it was a real secret such as password (which isn’t the case), how else should i pass it between jobs?</p>
<p>I’m using multiple envs and secrets, this is the only env that fails with that error.</p>
<p>the $SECRET is a simple json:<br>
{<br>
“AWS_ACCOUNT_ID”:“1234”<br>
}</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jonathan-be21">@jonathan-be21</a>,</p>
<p>I cannot reproduce the issue, please check my workflow: <a href="https://github.com/weide-zhou/ticket13/runs/902090365?check_suite_focus=true" rel="nofollow noopener">https://github.com/weide-zhou/ticket13/runs/902090365?check_suite_focus=true</a></p>
<p>If the output in step <code>Retrieve Secrets</code> doesn’t contain star *, which means it doesn’t contain the secrets. And the error will not happen.</p>
<p>Hence, could you please provide a sample repository for futher investigation?</p>
<p>Thanks</p>

**Comment by fmb-chin** [comment, marked as answer]
<p><a class="mention" href="/jonathan-be21">@jonathan-be21</a> I got this issue, too.<br>
But I found the root cause.</p>
<p>aws-actions/configure-aws-credentials@v1 will addMask for our aws accountid<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/b54a0c876df35708c179c98c4413c776161dff2a.png" data-download-href="/uploads/short-url/pRL4pZf2Ub9C9Kde61ksrRzNv7Q.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/44079276/181098985-e935add6-56f5-46ac-b801-8d4c4360b847.png" alt="image" data-base62-sha1="pRL4pZf2Ub9C9Kde61ksrRzNv7Q" width="690" height="232" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/b54a0c876df35708c179c98c4413c776161dff2a_2_690x232.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/b54a0c876df35708c179c98c4413c776161dff2a_2_1035x348.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/b/b54a0c876df35708c179c98c4413c776161dff2a_2_1380x464.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b54a0c876df35708c179c98c4413c776161dff2a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1410×476 40.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>you can use <code>mask-aws-account-id: 'no'</code> to avoid the issue.</p>
<pre><code class="lang-yaml">    - name: Configure AWS credentials
      uses: aws-actions/configure-aws-credentials@v1
      with:
        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
        aws-region: us-east-1
        mask-aws-account-id: 'no'
</code></pre>

**Reply by stephendarling** [reply]
Worked perfectly, thanks!

**Reply by ventz** [reply]
Thanks for sharing this! Wasted a lot of time trying to figure out why when passing a private AWS ECR string from one job to another, the output would not "make it over". What's worse, in the simple test cases it worked perfectly, due to configure-aws-credentials not being there. It turns out, the private ECRs contain the account #, and Github incorrectly filtered the string output as it was passed to another job. Posting this here in case it saves others some time in debugging this exact issue.

**Reply by MaNDiL555** [reply]
This fix the issue for aws-account-id but I'm facing the similar issue for other output
      REPO_NAME: ${{ env.REPO_NAME }}
      PROJECT_NAME: ${{ env.PROJECT_NAME }}

And I get 
Warning: Skip output 'REPO_NAME' since it may contain secret.
Warning: Skip output 'PROJECT_NAME' since it may contain secret.


Skip output 'REPO_NAME' since it may contain secret.

And most frustrating thing is that I have few more apps , with also same code, same action version and this happens only for this app

**Comment by ChenXiyu** [comment]
<p>Hi Folks, I got this problem too. And my output has nothing related to the AWS as well as any kind of secrets.</p>
<p>The result I got is: <code>Warning: Skip output 'images' since it may contain secret.</code>  and I cannot access the value from downstream jobs.</p>
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a>, I saw you asked a question:</p>
<blockquote>
<p>What’s display in the <code>log</code> for step “Retrieve Secrets”? Does it contain star ‘*’.</p>
</blockquote>
<p>The answer in my case is <strong>YES</strong>.<br>
The log looks like this:</p>
<pre><code class="lang-auto">echo "::set-output name=images::***/sub2conf:2349712309847"
</code></pre>
<p>And actually the value I set to the <code>images</code> is not a secret, the full value I gave is <code>94xychen/sub2conf:2349712309847</code></p>
<p>I guess this is because github detected it as secret by wrong, and I did not found any way to disable that functionality from both google and stackoverflow.</p>
<p>Am I missed something? Any hints you can provide?</p>
<p>For more informations:<br>
Here is the <a href="https://github.com/ChenXiyu/sub2conf/runs/2158918237?check_suite_focus=true" rel="noopener nofollow ugc">example</a></p>
<hr>

**Comment by ChenXiyu** [comment]
<p>Alright, I guess I found the issue.</p>
<p>The value <code>94xychen</code> been detected as a secret and been masked is because that it is the exactly same value with one of my repo level <code>secret</code>.</p>
<p>That is my docker-hub username, I putted it into my repo level secrets along with the docker-hub token.</p>
<p>I cannot tell whether this is a good idea or not, I just thought this mechanism may exposure secrets.</p>
<p>Anyway, I have changed my implementation from utilizing github output to use artifacts action, that make more sense in my case.</p>

**Comment by bob-bins** [comment]
<aside class="quote no-group" data-username="fmb-chin" data-post="9" data-topic="123791">
<div class="title">
<div class="quote-controls"></div>
 fmb-chin:</div>
<blockquote>
<p><code>configure-aws-credentials</code></p>
</blockquote>
</aside>
<p>Thank you so much for this. The <code>mask-aws-account-id</code> input is not documented in the current version of <code>aws-actions/configure-aws-credentials</code> so I spent hours trying to first figure out why my output kept reading as an empty string and then figure out how the heck to make this non-secret that github thinks is a secret to be passed as an output. Such a frustrating issue!</p>

**Comment by orihomie** [comment]
<p>Thanks for sharing! Have you worked around this somehow?</p>

**Comment by ChenXiyu** [comment]
<p>Yes, for me. Because the value that been masked is not really a secrete, So I just removed it from Github Secretes and putted it into repo directly.</p>

**Comment by ckingbailey** [comment]
<p>I’ve opened an <a href="https://github.com/aws-actions/configure-aws-credentials/issues/304" rel="noopener nofollow ugc">issue on the configure-aws-credentials repo</a> to request that this be documented</p>

**Comment by RichardTMiles** [comment]
<aside class="quote no-group" data-username="bob-bins" data-post="12" data-topic="123791">
<div class="title">
<div class="quote-controls"></div>
 bob-bins:</div>
<blockquote>
<p>aws-actions/configure-aws-credentials</p>
</blockquote>
</aside>
<p>This was also a solution for me, I’ve verified <a class="mention" href="/bob-bins">@bob-bins</a> in that no documentation is found on the readme. I have posted this quickly as suggestion to another README.md pull request on the repo.</p><aside class="onebox githubpullrequest" data-onebox-src="https://github.com/aws-actions/configure-aws-credentials/pull/430">
  <header class="source">

      <a href="https://github.com/aws-actions/configure-aws-credentials/pull/430" target="_blank" rel="noopener nofollow ugc">github.com/aws-actions/configure-aws-credentials</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/aws-actions/configure-aws-credentials/pull/430" target="_blank" rel="noopener nofollow ugc">Update README.md</a>
    </h4>

    <div class="branches">
      <code>aws-actions:master</code> ← <code>liwadman:master</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2022-04-26" data-time="22:53:30" data-timezone="UTC">10:53PM - 26 Apr 22 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/liwadman" target="_blank" rel="noopener nofollow ugc">
          <img alt="liwadman" src="https://user-images.githubusercontent.com/9538357/181098990-3f759dc4-f1f2-45c6-85d2-5ae638abd383.png" class="onebox-avatar-inline" width="20" height="20">
          liwadman
        </a>
      </div>

      <div class="lines" title="1 commits changed 1 files with 2 additions and 1 deletions">
        <a href="https://github.com/aws-actions/configure-aws-credentials/pull/430/files" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+2</span>
          <span class="removed">-1</span>
        </a>
      </div>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">Added explicit language that this action is not supported on github currently.
<span class="show-more-container"><a href="https://github.com/aws-actions/configure-aws-credentials/pull/430" target="_blank" rel="noopener nofollow ugc" class="show-more">…</a></span><span class="excerpt hidden">
*Issue #, if available:*

N/A

*Description of changes:*

I added clarifying language that says that the tagging with web identity tokens is not currently supported with github actions currently, as to not require customers to discover this themselves.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Many pull requests are open, not sure how active BDFL is.</p>
